SIG-Guest-Languages/JavaScript/guest-js-10-17-2024.md

```diff
@@ -0,0 +1,107 @@
+    * **Till:** Interesting idea, but it sounds like the same thing but based on P1 when it’s not
+    * **Calvin:** So it’s basically a module with the expectation of being a component
+    * **Till:** During Wizen-ining you can only use P1 APIs, then you get converted to P2. Beforehand the core module imports the P1 APIs, but you can’t use them (no host)
+* **Calvin:** Did you find Joel’s [component-init](https://github.com/dicej/component-init) useful?
+  * **Till:** Started looking into component init, wanted to use it for shared libs.
+  * Locally I have a branch for shared libraries, but doesn’t support init because component-init is a little difficult due to needing to duplicate a large amount of `componentize-py` (which is very python specific, and sometimes uses shared libs that aren’t likely to be present in other contexts)
+  * If it was in a more general form (i.e. run this, use these shared libraries & preopens) would have been easier to use, but was hard to pull out of `componentize-py`
+* **Till:** Has everyone seen the [dlopen() proposal](https://github.com/WebAssembly/component-model/issues/401)? Should be useful for python people (and is being built to be very easy to use in that sphere)
+  * **Dirk:** This *would* be very useful
+  * **Till:** Shouldn’t be too far from being implementable. Right now it passes bytes and gets a re
+* **Victor:** Does it make sense to prototype in the host?
+  * **Till:** the instantiation is really important and needs to be canonical
+  * **Till:** If someone wants to prototype this, we *could* at least try it, someone would have to implement it for `jco` and `wasmtime`.
+  * Moving from WIT to canonical builtins is often more challenging here.
+  * **Till:** reminder – Joel & I have been working on a new WASI HTTP extension proposal
+    * [Proposal](https://gist.github.com/tschneidereit/3b9aa3709ca8129ba6644fbaf8139fed)
+      * Summarized: a way to append to outgoing bodies, that enables streaming to happen in the backend, without WASM having to be invoked for every individual splice
+      * You can queue chunks to be written, exit the instance, and the host
+      * This has been very useful at Fastly
+      * Formal proposal incoming at a subgroup meeting
+    * [Joel’s example repo](https://github.com/dicej/wasi-http-tests)
+
+### Preview 3 (Yosh)
+
+* Receiving Qs from Microsoft: What are the things that need to happen to ship P3 sooner?
+  * Luke is updating the spec
+  * Joel is implementing Async & Errors
+* There are some open questions about the mapping of HTTP to P3 (can we reformulate?)
+  * **Till:** [https://github.com/WebAssembly/wasi-http/tree/main/wit-0.3.0-draft](https://github.com/WebAssembly/wasi-http/tree/main/wit-0.3.0-draft)
+* We can begin updating specs to have P3 drafts of existing specs
+  * Joel pulled out an RFC
+* The big open question is when is a ratified implementation likely (i.e. a second impl, `jco`)
+  * Right now it seems “Summer-ish” is the answer
```

Comment:
Northern hemisphere, I assume? I.e. ~2025Q3?

Reply:
Yeah, we are hoping sooner. The possible but aggressive goal is Q1 2025. But summer-ish would be end of Q2 or in Q3.